### DISCUSSION TITLE
How to enable multiple YAML frontmatters in Markdown?

### DISCUSSION THREAD

**Discussion by ChenZhongPu** [prettier/prettier#16720] [Q&A] [answered]
I am using [slidev](https://sli.dev/) in nvim. It seems that [prettier](https://www.lazyvim.org/extras/formatting/prettier) does not allow multiple YAML frontmatters in Markdown. 

For example,

```
---
layout: image
image: /path/to/the/image
backgroundSize: contain
---
```

will be formatted into:

```
---

layout: image
image: /path/to/the/image
backgroundSize: contain

---
```

But the former one is the way to set the layout in slidev. How to enable multiple front matters in Markdown?

**Comment by arham766** [comment, marked as answer]
There's no option for this. Prettier's markdown support only recognizes frontmatter at the very start of the file — that first `---` block gets extracted before the Markdown is parsed. Every later `---` pair is parsed as regular Markdown, where a `---` line is a thematic break, which is exactly the reformatting you're seeing. Per-slide frontmatter is a Slidev-specific extension, not part of any Markdown spec, so core Prettier won't handle it.

Two practical options:

1. Use `prettier-plugin-slidev` (on npm) — it teaches Prettier the Slidev dialect so the per-slide frontmatter blocks survive formatting.
2. Or add your slide decks to `.prettierignore` and keep Prettier away from them entirely.
